kratos/utilities/plane_approximation_utility.h

```diff
@@ -0,0 +1,244 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#if !defined(KRATOS_PLANE_APPROXIMATION_UTILITY )
+#define  KRATOS_PLANE_APPROXIMATION_UTILITY
+
+/* System includes */
+
+/* External includes */
+
+/* Project includes */
+#include "utilities/math_utils.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class PlaneApproximationUtility
+ * @ingroup KratosCore
+ * @brief Utility to compute an approximation plane for a set of points
+ * @details Provided a set of points, this utility computes the plane that best
+ * approximate them in a least square sense. The base point of the plane is the
+ * set of points average coordinates while the plane normal is taken as the
+ * eigenvector associated to the minimum eigenvalue of the matrix A. Matrix A is
+ * trans(P)*P, being P the matrix that stores in each row the difference between 
+ * the base point and each one of the points to approximate.
+ * @author Ruben Zorrilla
+ */
+class PlaneApproximationUtility
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of PlaneApproximationUtility
+    KRATOS_CLASS_POINTER_DEFINITION( PlaneApproximationUtility );
+    
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /**
+     * @brief Default constructor
+     */
+    PlaneApproximationUtility(){}
+
+    /// Destructor
+    virtual ~PlaneApproximationUtility(){}
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /**
+     * @brief Provided a list of point coordinates, computes 
+     * the best plane approximation in a least squares sense.
+     * @param rPointsCoords Vector containing the set of point coordinates
+     * @return rPlaneBasePointCoords Plane base point
+     * @return rPlaneNormal Plane normal
+     */
+    static void ComputePlaneApproximation(
+        const std::vector< array_1d<double,3> > &rPointsCoords,
+        array_1d<double,3> &rPlaneBasePointCoords,
+        array_1d<double,3> &rPlaneNormal)
+    {
+        GetPlaneBasePoint(rPointsCoords,rPlaneBasePointCoords);
+        GetPlaneNormal(rPointsCoords, rPlaneBasePointCoords, rPlaneNormal);
```

Comment:
Calculate would be better than Get for naming this methods to give a better impression of their cost. (Get gives the impression of returning an stored value)

Reply:
Ok. Done!.